features/step_definition/integration_steps.rb

```diff
@@ -13,10 +16,19 @@
 
   step 'Set regcode and url options'
 
-  @regcode = 'INVALID_REGCODE' unless options['regcode'] == 'VALID'
+  regcode = case options['regcode']
+  when 'INVALID'
+    'INVALID_REGCODE'
+  when 'EXPIRED'
+     @expired_regcode
+  when 'NOTYETACTIVATED'
+    @notyetactivated_regcode
+  when 'VALID'
+    @valid_regcode
+  end
```

Comment:
As a afterthought: `case` returns `nil` if none clauses match. Shouldn't you add
```ruby
else
  options['regcode']
```
branch to fallback to the proveded regcode if none matched?